File: VB/Solution2.Module.Win/Controllers/ReportCustomController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Reports.Win

Namespace Solution2.Module.Win.Controllers
	Partial Public Class ReportCustomController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
		End Sub
		Private reportService As WinReportServiceController
		Protected Overrides Sub OnFrameAssigned()
			MyBase.OnFrameAssigned()

			reportService = Frame.GetController(Of WinReportServiceController)()
			If reportService IsNot Nothing Then
				AddHandler reportService.CustomShowPreview, AddressOf reportService_CustomShowPreview
			End If
		End Sub
		Private Sub reportService_CustomShowPreview(ByVal sender As Object, ByVal e As CustomShowPreviewEventArgs)
			Dim coverPageReport As New XtraReport1()
			coverPageReport.CreateDocument()

			e.Report.CreateDocument()

			coverPageReport.Pages.AddRange(e.Report.Pages)
			coverPageReport.ShowPreviewDialog()

			e.Handled = True
		End Sub
	End Class
End Namespace

File: CS/Solution2.Module.Win/Controllers/ReportCustomController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using DevExpress.XtraReports.UI;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Reports.Win;

namespace Solution2.Module.Win.Controllers {
    public partial class ReportCustomController : ViewController {
        public ReportCustomController() {
            InitializeComponent();
            RegisterActions(components);
        }
        private WinReportServiceController reportService;
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();

            reportService = Frame.GetController<WinReportServiceController>();
            if(reportService != null) {
                reportService.CustomShowPreview += reportService_CustomShowPreview;
            }
        }
        private void reportService_CustomShowPreview(
            object sender, CustomShowPreviewEventArgs e) {
            XtraReport1 coverPageReport = new XtraReport1();
            coverPageReport.CreateDocument();

            e.Report.CreateDocument();

            coverPageReport.Pages.AddRange(e.Report.Pages);

            ReportPrintTool printTool = new ReportPrintTool(coverPageReport);
            printTool.ShowPreviewDialog();

            e.Handled = true;
        }
    }
}
